src/lib/market-service/idle/idle.ts

```diff
@@ -37,16 +38,23 @@ export class IdleMarketService extends CoinGeckoMarketService implements MarketS
 
     this.baseUrl = 'https://api.idle.finance'
     this.providerUrls = providerUrls
-    this.idle = axios.create({
-      timeout: 10000,
-      baseURL: this.baseUrl,
-      headers: {
-        Authorization:
-          'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjbGllbnRJZCI6IkFwcDIiLCJpYXQiOjE2NzAyMzc1Mjd9.pf4YYdBf_Lf6P2_oKZ5r63UMd6R44p9h5ybPprtJmT4',
+
+    this.idle = setupCache(
+      axios.create({
+        timeout: 10000,
+        baseURL: this.baseUrl,
+        headers: {
+          Authorization:
+            'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjbGllbnRJZCI6IkFwcDIiLCJpYXQiOjE2NzAyMzc1Mjd9.pf4YYdBf_Lf6P2_oKZ5r63UMd6R44p9h5ybPprtJmT4',
+        },
+      }),
+      {
+        // 5 seconds
+        ttl: 5 * 1000,
+        interpretHeader: false,
+        staleIfError: true,
```

Comment:
[staleIfError](https://axios-cache-interceptor.js.org/config/request-specifics#cache-staleiferror): `Enables cache to be returned if the response comes with an error, either by invalid status code, network errors and etc. You can filter the type of error that should be stale by using a predicate function.`.
Setting this to `true` so we consider responses stale on error. Equivalent to the previous `readOnError: false`.

Reply:
That's a neat flag.